diff --git a/target/linux/rockchip/image/armv8.mk b/target/linux/rockchip/image/armv8.mk
index cf2ef3145..d35d70292 100644
--- a/target/linux/rockchip/image/armv8.mk
+++ b/target/linux/rockchip/image/armv8.mk
@@ -64,6 +64,17 @@ define Device/armsom_sige5
 endef
 TARGET_DEVICES += armsom_sige5
 
+define Device/igkboard_rk3576
+  DEVICE_VENDOR := LingYun
+  DEVICE_MODEL := IGKBoard-RK3576
+  SOC := rk3576
+  DEVICE_DTS := igkboard-rk3576
+  UBOOT_DEVICE_NAME := sige5-rk3576
+  IMAGE/sysupgrade.img.gz := boot-common | boot-script | pine64-img | gzip | append-metadata
+  DEVICE_PACKAGES += u-boot-sige5-rk3576
+endef
+TARGET_DEVICES += igkboard_rk3576
+
 define Device/armsom_sige7
   DEVICE_VENDOR := ArmSoM
   DEVICE_MODEL := Sige7
